applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <caf/events/ble_common_event.h>
+
+#if CONFIG_DESKTOP_USB_ENABLE
+#include "usb_event.h"
+#endif
+
+#define MODULE dvfs
+#include <caf/events/module_state_event.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_DVFS_LOG_LEVEL);
+
+#include "ld_dvfs_handler.h"
+
+#define DVFS_RETRY_TIMEOUT		K_MSEC(500)
```

Comment:
Wouldn't it be better to use longer timeout if DVFS is uninitialized and shorter timeout after it reports busy? (e.g. after USB is connected we may need good performance earlier even if DVFS was busy)

Reply:
After an error we should wait for the work to perform retry (we should not retry instantly on application event).

Reply:
And how do you plan to do it in other way than the solution that was proposed previously?

Reply:
We could e.g. use `k_work_schedule` or rely on `atomic` to prevent rescheduling when already scheduled

Reply:
Ok I added two different timeouts